src/Language/PureScript/Sugar/Names/Env.hs

```diff
@@ -60,16 +60,16 @@ data Imports = Imports
   , importedModules :: S.Set ModuleName
   -- |
   -- The names of "virtual" modules that come into existence when "import as"
-  -- is used.
+  -- is used, mapped to the real module being imported.
   --
-  , importedVirtualModules :: S.Set ModuleName
+  , importedVirtualModules :: M.Map ModuleName ModuleName
```

Comment:
I'm not sure this is quite right actually, shouldn't it be `M.Map ModuleName [ModuleName]`? As you can have things like:

``` purescript
import A as X
import B as X
```

That makes the problem of figuring out where re-exports came from a little awkward again though, so maybe we actually need even more information in the RHS value (something like `Imports` again with qualified `X.foo` names pointing to the imported `A.foo`/`B.foo` names)